app/packages/addons-service/provider.ts

```diff
@@ -20,34 +30,67 @@ export default class Provider {
         this.node = node;
         this.currentUser = currentUser;
         this.getInternalUser();
+        this.getInternalResource();
+        this.getConfiguredStorageAddon();
         this.provider = provider;
     }
 
     async getInternalUser() {
         this.internalUser = await this.store.findRecord('internal-user', this.currentUser.user?.id);
     }
 
+    async getInternalResource() {
+        this.serviceNode = await this.store.findRecord('internal-resource', this.node.id);
+    }
+
+    async getConfiguredStorageAddon() {
+        if (this.serviceNode) {
+            const configuredStorageAddons = await this.serviceNode.get('configuredStorageAddons');
+            configuredStorageAddons.forEach((configuredAddon: ConfiguredStorageAddonModel) => {
+                if (configuredAddon.storageProvider.id === this.provider.id) {
+                    this.configuredStorageAddon = configuredAddon;
+                }
+            });
+        }
+    }
+
     async userAddonAccounts() {
         return await this.internalUser.authorizedStorageAccounts;
     }
 
-    createAccountForNodeAddon() {
-        return;
+    async createAccountForNodeAddon() {
+        const account = this.store.createRecord('authorized-storage-account', {
+            externalUserId: this.currentUser.user?.id,
+            externalUserDisplayName: this.currentUser.user?.fullName,
```

Comment:
I thiiink this comes from the OAuth account that gets linked after going through the process.